File: reactive-java/src/main/java/com/example/reactivejava/repository/EmployeeRepo.java
package com.example.reactivejava.repository;

import java.time.Duration;

import com.example.reactivejava.model.Employee;
import reactor.core.publisher.Mono;

import org.springframework.stereotype.Service;

@Service
public class EmployeeRepo {
	public Mono<Employee> findEmployeeById(String id){
		return Mono.just(id)
				.delayElement(Duration.ofMillis(1000L))
				.map(Employee::new);
	}
}

File: reactive-kotlin/src/main/kotlin/com/example/reactivekotlin/repository/EmployeeRepo.kt
package com.example.reactivekotlin.repository

import com.example.reactivekotlin.model.Employee
import org.springframework.stereotype.Service
import reactor.core.publisher.Mono
import reactor.kotlin.core.publisher.toMono
import java.time.Duration

@Service
class EmployeeRepo {

    fun findEmployeeById(id: String?): Mono<Employee> {
        return id.toMono()
            .delayElement(Duration.ofMillis(1000L))
            .map { Employee(id, "vineet") }
    }
}